pkg/sql/txn_restart_test.go

```diff
@@ -1708,27 +1708,76 @@ CREATE TABLE t.test (k INT PRIMARY KEY);
 				//   - a marker statement recognized by the filter.
 				//   - a write that will conflict with the read done by the filter and
 				//     will cause the txn's timestamp to be pushed.
-				if _, err := tx.Exec(
+				_, err = tx.Exec(
 					fmt.Sprintf(
 						`SHOW DATABASES; SELECT '%s'; INSERT INTO t.test VALUES (%d)`,
-						marker, i)); err != nil {
-					t.Fatal(err)
-				}
-
-				err = tx.Commit()
+						marker, i))
 				if moveOutOfAutoRetry {
 					if !isRetryableErr(err) {
 						t.Fatalf("expected retryable error, got: %v", err)
 					}
+					if err := tx.Rollback(); err != nil {
+						t.Fatal(err)
+					}
 				} else {
 					if err != nil {
 						t.Fatal(err)
 					}
+					if err := tx.Commit(); err != nil {
+						t.Fatal(err)
+					}
 				}
 			})
 	}
 }
 
+// TestPushedTxnRetriableError is just used to verify we should return an retry error
+// early to the client for explicit txn whose timestamp has been pushed by other txns.
+// We start two txns in this test, txn1 did a write first, then txn2 did a read, when txn1 try to
+// update the record that txn2 has read, it's timestamp will be pushed forward.
+// So txn1 will get a retriable error immediately.
```

Comment:
I have add a comment here about this test.

Reply:
Isn't this test testing the same thing as the one above, in the `moveOutOfAutRetry=true` case? I don't think it adds anything; I would remove it.